packages/frontend-2/components/viewer/Controls.vue

```diff
@@ -419,7 +422,12 @@ const measureShortcut = ref(
   `Measure mode ${getShortcutTitle(ViewerKeyboardActions.ToggleMeasurements)}`
 )
 
+const isTyping = computed(() => ui.threads.openThread.isTyping.value)
+
 const handleKeyboardAction = (action: ViewerKeyboardActions) => {
+  if (isTyping.value) {
```

Comment:
isTyping is not fully accurate - its got a debounce timeout of 7 seconds, so 7 seconds after you're done typing you're still shown as typing. Also you're shown as typing for the entire duration of the comment submission, so if you're on a slow internet connection that can block you as well. These kind of delays are bad UX imo and will cause users not to trust these shortcuts ("why tf isn't it working when it usually did??").

To properly fix this you'd probably need to introduce a new more accurate isTyping `isTypingRealtime` that has either no debounce at all or a very low one, and that only gets set/unset by typing, not by clicking submit in the comment editor.

While you're looking at this, also validate that `onKeyDownHandler` of `useIsTypingUpdateEmitter()` even works, I see that its attached to @keydown of `<ViewerCommentsEditor/>` but that component has no such event. So do double check what's going on there as well

Reply:
Thanks great feedback. 

A simpler solution is possibly blocking the shortcuts based on newThreadEditor rather than isTyping
<img width="1131" alt="image" src="https://github.com/user-attachments/assets/190e9f02-c48c-4241-85ab-1379d1b69d35">

Do you think this is an acceptable change to fix this, without the complications around debounce? I've testing and it's working, and I don't want to spend too long on this when I have some Workspaces work to do.